test/helpers/ethtxmanmock_e2e.go

```diff
@@ -17,9 +18,15 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+// Wrapper struct to associate a mutex with each simulated backend instance
+type SimulatedBackendWithMutex struct {
+	Backend *simulated.Backend
```

Comment:
Wdyt, maybe it is slightly easier, because you don't need to call `s.Backend.Client()` etc but use the shorthand (namely `s.Client()`)?

```suggestion
	*simulated.Backend
```

Reply:
done